SecretManager/src/V1/Client/BaseClient/SecretManagerServiceBaseClient.php

```diff
@@ -0,0 +1,674 @@
+ *
+ * Manages secrets and operations using those secrets. Implements a REST
+ * model with the following objects:
+ *
+ * * [Secret][google.cloud.secretmanager.v1.Secret]
+ * * [SecretVersion][google.cloud.secretmanager.v1.SecretVersion]
+ *
+ * This class provides the ability to make remote calls to the backing service through method
+ * calls that map to API methods.
+ *
+ * Many parameters require resource names to be formatted in a particular way. To
+ * assist with these names, this class includes a format method for each type of
+ * name, and additionally a parseName method to extract the individual identifiers
+ * contained within formatted names that are returned by the API.
+ *
+ * This class is currently experimental and may be subject to changes.
+ *
+ * @internal
+ *
+ * @method PromiseInterface accessSecretVersionAsync(AccessSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface addSecretVersionAsync(AddSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface createSecretAsync(CreateSecretRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface deleteSecretAsync(DeleteSecretRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface destroySecretVersionAsync(DestroySecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface disableSecretVersionAsync(DisableSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface enableSecretVersionAsync(EnableSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getIamPolicyAsync(GetIamPolicyRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getSecretAsync(GetSecretRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface getSecretVersionAsync(GetSecretVersionRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface listSecretVersionsAsync(ListSecretVersionsRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface listSecretsAsync(ListSecretsRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface setIamPolicyAsync(SetIamPolicyRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface testIamPermissionsAsync(TestIamPermissionsRequest $request, array $optionalArgs = [])
+ * @method PromiseInterface updateSecretAsync(UpdateSecretRequest $request, array $optionalArgs = [])
+ */
+class SecretManagerServiceBaseClient
+{
+    use GapicClientTrait;
+    use ResourceHelperTrait;
+
+    /** The name of the service. */
+    private const SERVICE_NAME = 'google.cloud.secretmanager.v1.SecretManagerService';
+
+    /** The default address of the service. */
+    private const SERVICE_ADDRESS = 'secretmanager.googleapis.com';
+
+    /** The default port of the service. */
+    private const DEFAULT_SERVICE_PORT = 443;
+
+    /** The name of the code generator, to be included in the agent header. */
+    private const CODEGEN_NAME = 'gapic';
```

Comment:
The tests are failing for PHP 5.6 because I forgot that private constants were added in PHP 7.1

Reply:
Ok this is our last note...I'm open to suggestions, but I'd like to avoid extra BC breaking changes down the line if possible.

Reply:
I think making the new clients 7.4+ is my preference, it'll just take some consideration

Reply:
So does that mean do nothing? Or convert these to private properties for the time being?

Reply:
We are gating on The Drop of PHP versions < 7.4 so there is no action needed here. Resolving.